### PR TITLE
docs: Add documentation for VFIO-AP passthrough

### PR DESCRIPTION
**Pull request by Jakob-Naucke**
Add a guide on how to pass a VFIO-AP device, such as Crypto Express cards on IBM Z mainframes, to a Kata container.
Like the documentation for VFIO-PCI, this was put in the virtcontainers README.

/cc @alicefr

Fixes: #658

### CODE REVIEW

File: src/runtime/virtcontainers/README.md [hunk 1]
```diff
@@ -320,6 +323,81 @@ Clear Containers kernel. If the driver is missing,  you can add it to your
...
+IBM Z mainframes (s390x) use the AP (Adjunct Processor) bus for their Crypto
+Express hardware security modules. Such devices can be passed over VFIO, which
+is also supported in Kata. Pass-through happens separated by adapter and
+domain, i.e. a passable VFIO device has one or multiple adapter-domain
+combinations.
+
+1. You must follow the [kernel documentation for preparing VFIO-AP passthrough](https://www.kernel.org/doc/html/latest/s390/vfio-ap.html).
+In short, your host kernel should have the following enabled or available as
+module (in case of modules, load the modules accordingly, e.g. through
+`modprobe`). If one is missing, you will have to update your kernel
```
Comment (devimc) [inline src/runtime/virtcontainers/README.md:337]: uhmm were you able to load a module in a container using `modprobe` ?

File: src/runtime/virtcontainers/README.md [hunk 2]
```diff
@@ -320,6 +323,81 @@ Clear Containers kernel. If the driver is missing,  you can add it to your
...
+correspondent VFIO device in `/dev/vfio`.
+
+```sh
+$ readlink /sys/devices/vfio_ap/matrix/a297db4a-f4c2-11e6-90f6-d3b88d6c9525/iommu_group
+../../../../kernel/iommu_groups/0
+$ ls /dev/vfio
+0 vfio
+```
+
+6. This device can now be passed. With `lszcrypt` from `s390-tools`, you can
```
Comment (devimc) [inline src/runtime/virtcontainers/README.md:388]: how can I install  `s390-tools` ?

File: src/runtime/virtcontainers/README.md (hunk 1 above)
Comment (Jakob-Naucke) [reply src/runtime/virtcontainers/README.md:337]: > In short, your host kernel should have the following enabled

No, this is host kernel

File: src/runtime/virtcontainers/README.md (hunk 2 above)
Comment (Jakob-Naucke) [reply src/runtime/virtcontainers/README.md:388]: [s390-tools](https://github.com/ibm-s390-tools/s390-tools) is an IBM set of tools for Linux on Z mainframes. It's packaged as `s390-tools` on most distributions, on Red Hat-based it's `s390utils`. Should that be mentioned explicitly?

Comment (devimc) [reply src/runtime/virtcontainers/README.md:388]: yes please,

File: src/runtime/virtcontainers/README.md (hunk 1 above)
Comment (devimc) [reply src/runtime/virtcontainers/README.md:337]: ahh sorry , I missed the word `host`, thanks

File: src/runtime/virtcontainers/README.md (hunk 2 above)
Comment (Jakob-Naucke) [reply src/runtime/virtcontainers/README.md]: done

**Review by bergwolf — APPROVED** [review]
Thanks!

**Pull request merged**